File: archives/C#/0049. Group Anagrams.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        var result=new List<IList<string>>();
        var rep=new Dictionary<string,List<string>>();
        foreach(var str in strs){
            string sortedStr=SortString(str);
            if(!rep.ContainsKey(sortedStr)){
                rep[sortedStr]=new List<string>{str};
            }
            else{
                rep[sortedStr].Add(str);
            }
        }
        foreach(var k in rep.Keys){
            result.Add(rep[k]);
        }
        return result;
    }
    public string SortString(string str){
        var curList=str.ToCharArray();
        Array.Sort(curList);
        string rep="";
        foreach(var c in curList){
            rep+=c;
        }
        return rep;
    }
}

File: archives/20190519python/0049. Group Anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dict_strs={}
        for ss in strs:
            cur_str_list=[x for x in ss]
            cur_str_list.sort()
            cur_str=str(cur_str_list)
            
            if cur_str not in dict_strs:
                dict_strs[cur_str]=[ss]
            else:
                dict_strs[cur_str].append(ss)
                
        rep=[]
        for key in dict_strs:
            rep.append(dict_strs[key])
        
        return rep
